okteto/okteto
# Use dynamic kube token for k8s requests

# Proposed changes

Fixes https://github.com/okteto/app/issues/6906

~This branch requires this PR to be merged first: https://github.com/okteto/okteto/pull/3888~ ✅  done

## Description

This PR introduces a request to fetch the dynamic kubetoken from the Okteot backend. 

The dynamic kubetoken will be the default behaviour. If the endpoint is not available, the CLI will continue working with the static token returned in the graphql API call to retrieve credentials.

As part of this PR I've added an e2e test for `okteto kubetoken` which checks that the token returned is a dynamic token (with expiration date set)

## How to test

1. Build this branch and try  to deploy a project such as https://github.com/okteto/movies
2. Everything should work without any noticable effect
3. Now change the `kubetokenPathTemplate` and make the path such as `"%s/auth/kubetoken-notfound/%s"`
4. Build the CLI and try to deploy the same project
5. Everything should work without any noticable effect
6. Check the debug logs of both executions and make sure only in the second execution (with the broken path) you get the following log: `DEBU[0000] Dynamic Kubetoken not available, falling back to static token`
